pkg/kubelet/images/image_manager.go

```diff
@@ -173,13 +174,14 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 // applyDefaultImageTag parses a docker image string, if it doesn't contain any tag or digest,
 // a default tag will be applied.
 func applyDefaultImageTag(image string) (string, error) {
-	named, err := dockerref.ParseNormalizedNamed(image)
+	_, tag, _, err := parsers.ParseImageName(image)
 	if err != nil {
 		return "", fmt.Errorf("couldn't parse image reference %q: %v", image, err)
 	}
-	_, isTagged := named.(dockerref.Tagged)
-	_, isDigested := named.(dockerref.Digested)
-	if !isTagged && !isDigested {
+	if strings.Contains(image, ":latest") {
```

Comment:
should this be more specific to looking at the end of the image, and in not taking dependencies on the default tag chosen in `parsers.ParseImageName`

```
if len(digest) == 0 && len(tag) > 0 && !strings.HasSuffix(image, ":"+tag) {
		// we just concatenate the image name with the default tag here instead ...
		image = image + ":" + tag
}
```

Reply:
good call. updated.